flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CastFunctionMiscLegacyITCase.java

```diff
@@ -51,11 +52,12 @@ Stream<TestSetSpec> getTestSetSpecs() {
                                 BuiltInFunctionDefinitions.CAST, "legacy cast failure returns null")
                         .onFieldsWithData("invalid")
                         .andDataTypes(STRING().notNull())
-                        .testSqlRuntimeError(
-                                "CAST(f0 AS BIGINT)",
-                                "Column 'EXPR$0' is NOT NULL, however, a null value is "
-                                        + "being written into it. You can set job configuration "
-                                        + "'table.exec.sink.not-null-enforcer'='DROP' to suppress "
-                                        + "this exception and drop such records silently."));
+                        .testSqlResult("CAST(f0 AS BIGINT)", null, BIGINT()),
+                TestSetSpec.forFunction(
+                                BuiltInFunctionDefinitions.CAST, "legacy cast failure returns null")
+                        .onFieldsWithData("100")
+                        .andDataTypes(STRING().notNull())
+                        .testSqlResult(
+                                "CAST(CAST(f0 AS DECIMAL(2, 0)) AS STRING)", null, STRING()));
```

Comment:
IIUC, you only tested runtime issues and did not test for constant folding issues, right?

Reply:
it is not important, the original ScalarFunctionTests all using constant folding